rpc_component/schemata.py

```diff
@@ -134,20 +135,59 @@ def is_version_ids_unique(releases):
     }
 )
 
-component_metadata_schema = Schema(
+dependencies_schema = Schema(
     {
-        "dependencies": And(
+        Optional("dependencies"): And(
             [
                 {
                     "name": And(str, len),
                     "constraints": constraints_schema,
                 },
             ],
             is_value_unique("name"),
+        )
+    }
+)
+
+artifacts_file_schema = Schema(
+    {
+        "type": "file",
+        "source": And(str, len),
+        Optional("dest"): And(str, len),
+        Optional("expire_after"): And(str, len)
+    }
+)
+
+artifacts_log_schema = Schema(
+    {
+        "type": "log",
+        "source": And(str, len)
+    }
+)
+
+artifacts_schema = Schema(
+    {
+        Optional("artifacts"): And(
+            [
+                Or(artifacts_file_schema, artifacts_log_schema, None)
```

Comment:
What does `None` represent? How does that compare with an empty list?

Reply:
``None`` is to allow for the list to be empty, this this is to cater for:

```yaml
artifacts: []
```

Reply:
Looks like it works without it, so I've removed it.